x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/query/KqlQueryBuilder.java

```diff
@@ -97,7 +97,7 @@ public static KqlQueryBuilder fromXContent(XContentParser parser) {
 
     @Override
     public TransportVersion getMinimalSupportedVersion() {
-        return TransportVersions.KQL_QUERY_ADDED;
+        return TransportVersions.KQL_QUERY_TECH_PREVIEW;
```

Comment:
Why do we need a new `TransportVersion`? Shouldn't the previous one be enough?

Reply:
Serveless have already the TransportVersions corresponding to KQL_QUERY_ADDED deployed but do not have the corresponding query because it was added only for snapshot build.

Reply:
But the QueryBuilder was always added and did not depend on snapshot builds, is that correct? So there is no need to change the `getMinimalSupportedVersion`, as the query is already available?

I'm confused because there is no change on the serialization formats that should imply a new `TransportVersion`.

Reply:
Nope, the old code was adding the QueryBuilder only for snapshot builds:

```java
        if (Build.current().isSnapshot()) {
            return List.of(new SearchPlugin.QuerySpec<>(KqlQueryBuilder.NAME, KqlQueryBuilder::new, KqlQueryBuilder::fromXContent));
        }
```

So it is not yet available in serverless and I think we need to update the `getMinimalSupportedVersion` accordingy now that the conditional has been removed.